ott/geometry/costs.py

```diff
@@ -90,17 +90,88 @@ def tree_unflatten(cls, aux_data, children):
     return cls(*children)
 
 
+@jax.tree_util.register_pytree_node_class
+class RBFCost(CostFn):
+  """A radial-basis function cost class for translation invariant costs.
+
+  Such costs are defined as
+
+  c(x,y) = h(z), where z := x-y.
+
+  where h is a function strictly convex (or concave) function mapping vectors
+  to real-values.
+
+  For completeness (and differentiation using the Brenier theorem), the user
+  is also supposed to provide the Legendre transform of `h`, whose gradient (the
+  inverse of the gradient of `h`) will be used to form a Brenier map.
+  """
+
+  def h(self, z: jnp.ndarray) -> float:
+    pass
+
+  def h_legendre(self, z: jnp.ndarray) -> float:
+    pass
+
+  def pairwise(self, x: jnp.ndarray, y: jnp.ndarray) -> float:
+    """Evaluate h on difference between x and y."""
+    return self.h(x - y)
+
+  def tree_flatten(self):
```

Comment:
Would remove the `tree_{un,}flatten` from this class, since it will be abstract anyways.
Alt. is to only keep the `tree_unflatten` as the most generic variant with `cls(*children, **aux_data)`, but I prefer the former.

Reply:
This makes a lot of sense, thanks